procfs/cmdline.go

```diff
@@ -184,6 +184,40 @@ func (c *Cmdline) SetAll(args []string) {
 	}
 }
 
+// Delete deletes a kernel parameter by exact match.
+func (c *Cmdline) Delete(v *Parameter) {
+	c.Lock()
+	defer c.Unlock()
+
+	for i, value := range c.Parameters {
+		if value.key == v.key {
+			for _, val := range v.values {
+				for j, v := range value.values {
+					if v == val {
+						c.Parameters[i].values = append(c.Parameters[i].values[:j], c.Parameters[i].values[j+1:]...)
```

Comment:
we could make it less weird looking by using e.g. https://pkg.go.dev/slices#Delete

Reply:
or even `DeleteFunc` there

Reply:
oh, this is cool